pkg/iterator/iter.go

```diff
@@ -104,6 +104,9 @@ type Args struct {
 	// If unspecified the default for the given client will be used.
 	ServiceAccount string
 
+	// ServiceAccountNamespace is the namespace of the ServiceAccount.
+	ServiceAccountNamespace string
```

Comment:
1. This is optional and should be stated so.  I suggest combining the comment for the two as follows:
```
	// Specify the ServiceAccount object used to construct a security token
	// with the audience string from the SnapshotMetadataService CR.
	// If either of the following fields are unspecified, the default for the given client will be used.
	ServiceAccountNamespace string
	ServiceAccount string
```
It would be good if `ServiceAccount` was renamed to `ServiceAccountName`.

2. The `args.Validate` function should be enhanced to ensure that either both fields are empty or both are set - impacts validation UTs too.